app/src/main/java/org/wikipedia/talk/TalkTopicsActivity.kt

```diff
@@ -145,6 +145,8 @@ class TalkTopicsActivity : BaseActivity() {
 
                     pageTitle = PageTitle(newNamespace, StringUtil.removeNamespace(pageTitle.prefixedText),
                             WikiSite.forLanguageCode(WikipediaApp.getInstance().language().appLanguageCodes[pos]))
+
+                    talkTopicsProvider = TalkTopicsProvider(pageTitle)
```

Comment:
Should we `cancel()` the previous provider before creating a new one?

Reply:
Yes we should. Done.